source/common/common/utility.cc

```diff
@@ -486,6 +486,33 @@ std::string StringUtil::removeCharacters(const absl::string_view& str,
   return absl::StrJoin(pieces, "");
 }
 
+absl::optional<uint64_t> StringUtil::readAndRemoveLeadingDigits(absl::string_view& s) {
+  absl::optional<uint64_t> output;
+  const char* p = s.data();
+  const char* limit = p + s.size();
+  uint64_t val = 0;
+  while (p < limit) {
+    const char c = *p;
+    if (c < '0' || c > '9') {
+      break;
+    }
+    uint64_t new_val = (val * 10) + (c - '0');
+    if (new_val / 8 < val) {
+      // Overflow occurred
+      return output;
```

Comment:
I think it would be clearer to return nullopt.

Reply:
In fact, the 'output' variable doesn't seem worthwhile--each of its uses would be better if replaced by either "return nullopt" or "return val".